### PR TITLE
[WISE-8] Assets with 0 balance

### PR DESCRIPTION
**Pull request by omar-pravica-io**

### CODE REVIEW

**Comment by ma-aboelenein** [comment]
the assets of the previous account persists if the account you switched to doesn't have assets in them

File: src/hooks/useAssets/assetsStore.tsx [hunk 1]
```diff
@@ -86,7 +94,9 @@ export const assets = atom(async get => {
       icon: StxTokenIcon,
       amount: valueFromBalance(stxBalance as BigNumber, 'stx'),
     };
-    results.push(stxToken);
+    if (stxBalance > new BigNumber(0)) {
```
Comment (ma-aboelenein) [inline src/hooks/useAssets/assetsStore.tsx:97]: `new BigNumber(0)` should be extracted into a constant with a descriptive name

File: src/hooks/useAssets/assetsStore.tsx [hunk 2]
```diff
@@ -29,38 +29,46 @@ export const assets = atom(async get => {
...
-            contractId,
+      Object.keys(assets)
+        .filter(fungibleKey => {
+          const assetToken = assets[fungibleKey] as FungibleResponse &
+            NonFungibleResponse;
+          const amount = assetToken.balance || assetToken.count;
+
+          return amount > new BigNumber(0);
+        })
+        .map(async fungibleKey => {
```
Comment (ma-aboelenein) [inline src/hooks/useAssets/assetsStore.tsx:40]: the concatenation of a `filter` and `map` functions can be combined in one `reduce` function or with a `foreach` and an empty array placeholder

File: src/hooks/useAssets/assetsStore.tsx [hunk 3]
```diff
@@ -29,38 +29,46 @@ export const assets = atom(async get => {
     isFungible = false,
   ) => {
     return Promise.all(
-      Object.keys(assets).map(async fungibleKey => {
```
[the remark this answers is not in the document]
Comment (omar-pravica-io) [reply src/hooks/useAssets/assetsStore.tsx:32]: done

File: src/hooks/useAssets/assetsStore.tsx (hunk 1 above)
Comment (omar-pravica-io) [reply src/hooks/useAssets/assetsStore.tsx]: done

File: src/hooks/useAssets/assetsStore.tsx (hunk 2 above)
Comment (omar-pravica-io) [reply src/hooks/useAssets/assetsStore.tsx]: done

**Pull request merged**